lib/src/http_session.dart

```diff
@@ -61,34 +67,34 @@ class HttpSession implements IOClient {
     Object? body,
     Encoding? encoding,
   }) async {
-    return _updateResponse(await _httpDelegate.delete(
+    Map<String, String> headers0 = {
+      "Cookie": _getCookieHeader(url.host, url.path)
+    };
+    headers0.addAll(headers ?? {});
```

Comment:
Do you think this code should be rewrote as a function so we can reuse it and also easier to maintain?

Reply:
I agree but I couldn't come up with a good way to do this, any ideas?

Reply:
We can do something like this:

``` dart
  Map<String, String> _getHeaders(Uri url, Map<String, String> headers) {
    Map<String, String> tempHeaders = {
      "Cookie": _getCookieHeader(url.host, url.path)
    };
    tempHeaders.addAll(headers ?? {});
    return tempHeaders;
  }
```

Then we can pass the `_getHeaders(url, headers)` to all `headers` parameters.

Reply:
I ended up rewriting that bit when trying to get it to support redirects ([a2ffb8d](https://github.com/lamnhan066/http_session/pull/5/commits/a2ffb8d3419d04ff2ef405990b92b937495431ee))

Lmk what you think please

Reply:
Can we use `_sendRequest` for this `delete` method?

Reply:
Oops, fixed